src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,149 @@
+            tail.next = newNode;
+            newNode.prev = tail;
+            tail = newNode;
+        } else {
+            Node<T> current = getNodeByIndex(index);
+            newNode.prev = current.prev;
+            newNode.next = current;
+            current.prev.next = newNode;
+            current.prev = newNode;
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        if (list != null && !list.isEmpty()) {
+            for (T element : list) {
+                Node<T> newNode = new Node<>(tail, element, null);
+                if (tail != null) {
+                    tail.next = newNode;
+                }
+                tail = newNode;
+                if (size == 0) {
+                    head = newNode;
+                }
+                size++;
+            }
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndexElement(index);
+        Node<T> current = getNodeByIndex(index);
+        return current.item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndexElement(index);
+        Node<T> current = getNodeByIndex(index);
+        T oldItem = current.item;
+        current.item = value;
+        return oldItem;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndexElement(index);
+        Node<T> removed = getNodeByIndex(index);
+        removeItem(removed);
+        return removed.item;
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> current = head;
+        while (current != null) {
+            if ((object == null && current.item == null) || (object != null
+                    && object.equals(current.item))) {
+                removeItem(current);
+                return true;
+            }
+            current = current.next;
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return this.size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private static class Node<T> {
+        private T item;
+        private Node<T> next;
+        private Node<T> prev;
+
+        public Node(Node<T> prev, T item, Node<T> next) {
+            this.prev = prev;
+            this.item = item;
+            this.next = next;
+
+        }
+    }
+
+    private boolean getIndexElement(int index) {
+        return index >= 0 && index < size;
+    }
+
+    private Node<T> getNodeByIndex(int index) {
+        Node<T> current = head;
+        for (int i = 0; i < index; i++) {
+            current = current.next;
+        }
+        return current;
+    }
+
+    private void checkIndexElement(int index) {
+        if (!getIndexElement(index)) {
+            throw new IndexOutOfBoundsException("Invalid index" + index);
+        }
+    }
+
+    private void removeItem(Node<T> node) {
```

Comment:
since item is also a name of the field where we store value of the node I'd recommend changing method name to something else to avoid any confusion between them
```suggestion
    private void unlink(Node<T> node) {
```